src/v/kafka/server/usage_manager.cc

```diff
@@ -256,15 +257,48 @@ ss::future<> usage_manager::accounting_fiber::stop() {
     }
 }
 
-ss::future<> usage_manager::accounting_fiber::close_window() {
+void usage_manager::accounting_fiber::close_window() {
     const auto now = ss::lowres_system_clock::now();
-    _buckets[_current_window].u = co_await _um.map_reduce0(
-      [](usage_manager& um) { return um.sample(); },
-      _buckets[_current_window].u,
-      [](const usage& acc, const usage& x) { return acc + x; });
-    _buckets[_current_window].end = epoch_time_secs(now);
+    const auto now_ts = epoch_time_secs(now);
+    const auto before_close_idx = _current_window;
+    _buckets[before_close_idx].end = now_ts;
     _current_window = (_current_window + 1) % _buckets.size();
     _buckets[_current_window].reset(now);
+    /// The timer must progress so subsequent windows may be closed, async work
+    /// may hold that up for a significant amount of time, therefore async work
+    /// will be dispatched in the background and the result set written to the
+    /// correct bucket when it is eventually computed
```

Comment:
i guess i'm unsure why we need to worry about writing back into the "correct" bucket. i mean, what makes the start of the collection of stats better than the point in time when we have an answer?

Reply:
This is more to prevent a bunch of buckets opening and losing historical data in the event there are many crashes in a row and the disk write interval is short enough to persist those open buckets to disk.

This is a best effort sort of thing. In the case it isn't immediately obvious which bucket to re-open, the logic reverts to just opening a new one anyway.